hugo/content/documentation/administration/acl/index.md

```diff
@@ -0,0 +1,250 @@
+| `@in`    | In this channel                      |  |
+| `@out`   | Outside of this channel              |  |
+| `@sub`   | Sub-channel - `@sub,a,b,c`           |  |
+
+#### `in` Group
+
+Given a channel tree
+
+```text
+Root
+  ChanA1
+    ChanA11
+    ChanA12
+  ChanB
+```
+
+To deny users outside of ChanA1 to send text messages to it, but allow users inside of it to send text messages, we add two rules.
+
+* *Deny* *Text message* to `@all`
+* *Allow* *Text message* to `@in`
+
+Now users in `Root`, `ChanA11`, `ChanA12`, ChanB can not send text messages to ChanA1. But users inside of ChanA1 can.
+
+![Screenshot of in rule example all rule](acl-in-example1-1all.jpg)
+
+![Screenshot of in rule example in rule](acl-in-example1-2in.jpg)
+
+The rule is inherited into `ChanA11` and `ChanA12`. But it applies as a rule within that channel. So users outside of `ChanA11` can not send text messages to `ChanA11`, but users inside of it can. Users outside of `ChanA12` can not send text messages to `ChanA12`, but users inside of it can.
+
+If we define the second rule [on `ChanA1`] as `@~in` instead of `@in` then the inherited rule in `ChanA11` and `ChanA12` is interpreted in the context of `ChanA1`. That means that a user in `ChanA1` will be able to send text messages to `ChanA1`, `ChanA11`, and `ChanA12`. But a user not in `ChanA1` can not. Users in `ChanA11` and `ChanA12` will also not be able to send text messages to `ChanA1`, `ChanA11`, and `ChanA12`.
+
+#### `sub` Group
+
+The `sub` group applies to subchannels and has three optional parameters `a,b,c`. It is written as `@sub` or `@sub,a` or `@sub,a,b` or `@sub,a,b,c` where `a`, `b`, and `c` are integer numbers.
+
+| Parameter | Description                      | Default |
+| --------- | -------------------------------- | ------- |
+| a         | minimum number of common parents | `0`     |
+| b         | minimum path depth               | `1`     |
+| c         | maximum path depth               |         |
+
+Given the channel tree
+
+```text
+ChanA
+  ChanB
+    ChanC
+```
+
+| Selector     | On channel | Applies to user in        |
+| ------------ | ---------- | ------------------------- |
+| `@sub,0,0`   | `ChanA`    | `ChanA`, `ChanB`, `ChanC` |
+| `@sub,0,1`   | `ChanA`    |          `ChanB`, `ChanC` |
+| `@sub,0,2`   | `ChanA`    |                   `ChanC` |
+| `@sub,0,0,0` | `ChanA`    | `ChanA`                   |
+| `@sub,0,0,1` | `ChanA`    | `ChanA`, `ChanB`          |
+| `@sub,0,0,2` | `ChanA`    | `ChanA`, `ChanB`, `ChanC` |
+| `@sub,0,1,1` | `ChanA`    |          `ChanB`          |
+| `@sub,0,2,2` | `ChanA`    |                   `ChanC` |
+
+In other words, `b` and `c` define the child depth span (children of `b`-th to `c`-th depth).
+
+##### `sub` Example 1
+
+Given the channel tree
+
+```text
+ChanA
+  ChanB
+    ChanC
+```
+
+A `@sub` rule (equivalent to `@sub,0,1`) on `ChanA` applies to users in `ChanB` and `ChanC` but gives permission on `ChanA`.

```

Comment:
The `sub` rule only makes sense when using the `~` modifier. Otherwise, there will never be any children or parents that are considered as the context in which the ACL is evaluated in is always the current channel.

Reply:
I added the tilde to all `@sub` mentions. Not just this commented one. I hope that is correct.

And I added a paragraph explaining what you told me here.